### PR TITLE
Offline storage documentation for iOS and Android

### PR DESCRIPTION
**Pull request by mbruin-NR**
Added needed documentation for configuring offline storage for iOS and Android

### CODE REVIEW

**Comment by akristen** [comment]
Hey @mbruin-NR ! This is top priority tomorrow. Will be getting this reviewed + out before EOD. Thanks for sending this PR through

**Comment by akristen** [comment]
Hey @mbruin-NR . Looks like our recent migration to Netlify might've done something wonky with your fork. Our eng team recommended that you rebase your fork, then update this branch so it builds. If you need help with that, reach out to the hero in our Docs channel and we'll get you sorted.

this PR has been reviewed, though. As soon as the required checks go through, I'll get it merged.

**Review by akristen — DISMISSED** [review]

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx [hunk 1]
```diff
@@ -433,6 +433,37 @@ To change settings, you have two options (if the setting supports it):
         </tr>
 </table>
 
+## Offline Storage [#dt]
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx:436]:
```suggestion
## Offline storage [#android-offline-storage]
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx [hunk 2]
```diff
@@ -433,6 +433,37 @@ To change settings, you have two options (if the setting supports it):
...
+            </th>
+            <th>
+                Example
+            </th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+                Enable or disable the storing of data payloads in the file system when an internet connection is not detected.
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx:452]:
```suggestion
                Enable or disable offline data storage when no internet connection is available.
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx [hunk 3]
```diff
@@ -433,6 +433,37 @@ To change settings, you have two options (if the setting supports it):
...
+                Example
+            </th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+                Enable or disable the storing of data payloads in the file system when an internet connection is not detected. 
+                
+                * To configure the amount of offline storage refer to [Set max offline storage size](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage).
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx:454]:
```suggestion
                * To configure the amount of offline storage, see [Set max offline storage size](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage).
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx [hunk 4]
```diff
@@ -547,32 +576,43 @@ If you modify any of crash and error reporting settings below, be sure to call t
...
+                Default value is `mobile-crash.newrelic.com`.
 
-                Default value is `true`.
             </td>
             <td>
                 ```swift
-                NRFeatureFlag_HandledExceptionEvents(false)
+                 NewRelic.start(withApplicationToken:YOUR_APP_TOKEN 
+                        andCollectorAddress:"harvest-upload.domain.com" 
+                        andCrashCollectorAddress:"crash-upload.domain.com");
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx:606]:
```suggestion
                NewRelic.start(withApplicationToken:YOUR_APP_TOKEN 
                       andCollectorAddress:"harvest-upload.domain.com" 
                       andCrashCollectorAddress:"crash-upload.domain.com");
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx [hunk 5]
```diff
@@ -547,32 +576,43 @@ If you modify any of crash and error reporting settings below, be sure to call t
...
                 ```
             </td>
         </tr>
     </tbody>
 </table>
 
-## Data endpoint settings [#logging]
+## Crash and error reporting feature flag [#crashes]
+
+If you modify any of crash and error reporting settings below, be sure to call the feature flag before the New Relic iOS agent start call.
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx:615]:
```suggestion
If you modify any of crash and error reporting settings below, be sure to call the feature flag before the iOS agent start call.
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx [hunk 6]
```diff
@@ -915,7 +950,37 @@ If you modify any of settings below, be sure to call the feature flag before the
...
                 ```swift
-                NRFeatureFlag_FedRampEnabled(true)
+                NewRelic.enableFeatures(NRMAFeatureFlags.NRFeatureFlag_FedRampEnabled)
+                ```
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Offline Storage [#dt]
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx:960]:
```suggestion
## Offline storage [#ios-offline-storage]
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx [hunk 7]
```diff
@@ -915,7 +950,37 @@ If you modify any of settings below, be sure to call the feature flag before the
...
+            </th>
+            <th>
+                Example
+            </th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+                Enable or disable the storing of data payloads in the file system when an internet connection is not detected. To configure the amount of offline storage refer to [Set max offline storage size](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage).
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings.mdx:976]:
```suggestion
                Enable or disable offline data storage when no internet connection is available. To configure the amount of offline storage, see [Set max offline storage size](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage).
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage.mdx [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+  - Custom instrumentation
+metaDescription: Mobile app monitoring API used to set the maximum size of offline storage.
+freshnessValidatedDate: 2024-04-19
+---
+
+Sets the maximum size of total data that can be stored for offline storage.
+
+By default, mobile monitoring can collect a maximum of 100 megaBytes of offline storage. When a data payload fails to send because the device doesn't have an internet connection, it can be stored in the file system until an internet connection has been made. After a typical harvest payload has been successfully sent, all offline data is sent to New Relic and cleared from storage.
+
+This method controls the maximum amount of offline storage that can be stored in the file system. When the size limit is reached, the New Relic agent will discard the new payloads that couldn't be stored. This method lets you override the maximum size of that offline storage.
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage.mdx:17]:
```suggestion
This method controls the maximum amount of offline storage that can be stored in the file system. When the size limit is reached, the New Relic mobile agent will discard the new payloads that couldn't be stored. This method lets you override the maximum size of that offline storage. 
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage.mdx [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+Sets the maximum size of total data that can be stored for offline storage.
+
+By default, mobile monitoring can collect a maximum of 100 megaBytes of offline storage. When a data payload fails to send because the device doesn't have an internet connection, it can be stored in the file system until an internet connection has been made. After a typical harvest payload has been successfully sent, all offline data is sent to New Relic and cleared from storage.
+
+This method controls the maximum amount of offline storage that can be stored in the file system. When the size limit is reached, the New Relic agent will discard the new payloads that couldn't be stored. This method lets you override the maximum size of that offline storage. 
+
+<Callout variant="important">
+  Offline storage is enabled by default but can be disabled with a feature flag using the `disableFeatures` function. See [Configure mobile monitoring settings](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings) to disable offline storage.
+</Callout>
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage.mdx:21]:
```suggestion
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage.mdx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+### Kotlin [#kotlin]
+
+```kotlin
+NewRelic.setMaxOfflineStorageSize(1000)
+```
+
+        </TabsPageItem>
+        <TabsPageItem id="ios">
+
+## Syntax [#syntax]
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage.mdx:102]:
```suggestion
<Callout variant="important">
Offline storage is enabled by default but can be disabled with a feature flag using the `disableFeatures` function. For details see [Offline storage](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings/#ios-offline-storage].
</Callout>

## Syntax [#syntax]
```

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage.mdx [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+            Android
+        </TabsBarItem>
+        <TabsBarItem id="ios">
+            iOS
+        </TabsBarItem>
+    </TabsBar>
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
```
Comment (ally-sassman) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-offline-storage.mdx:35]:
```suggestion
        <TabsPageItem id="android">
        
<Callout variant="important">
Offline storage is enabled by default but can be disabled with a feature flag using the `disableFeatures` function. For details see [Offline storage](/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/configure-settings/#android-offline-storage].
</Callout>

## Syntax [#syntax]
```

**Review by ghost — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**